src/facade/reply_builder.h

```diff
@@ -298,6 +312,46 @@ class RedisReplyBuilder : public SinkReplyBuilder {
   bool is_resp3_ = false;
 };
 
+// Redis reply builder interface for sending RESP data.
+class RedisReplyBuilder2Base : public SinkReplyBuilder2 {
```

Comment:
Crucial difference, all the SendArray, SendScoredArray, SendMget, etc.. are **not** part of the base interface - it represents just the essential redis protocol types. Big functions are implemented as helper functions. Keeping the base interface simple allows us to remove a lot of code in HttpSerializer/EvalSerializer/CapturingReplyBuilder that was handling all of this stuff manually

Reply:
Are you saying that knowing the arrays in advance does not provide any performance benefits?

Reply:
Knowing the lenghts can be an interesting idea for further optimizations. However we don't use that info even as of now. I only generalized the approach from your SendMget/ScoredArray optimizations, so it can be triggered via a manual calls to StartArray and SendBulkString

Reply:
#3481 is the extension